### ISSUE TITLE
no-cond-assign giving error on intentional assignment in for loop

### ISSUE THREAD

**Issue by carlbennettnz** [eslint/eslint#4023]
```js
/* eslint no-cond-assign: 2 */
for (var i = 0, item; (item = arr[i]); i++) { }
```

Even with the `no-cond-assign` set to the default `except-parens` rule, this still fails. Intentional?

Output:

```
  2:1   error  Expected a conditional expression and instead saw an assignment  no-cond-assign
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Can you provide version of ESLint you are using and your config file please?

**Comment by carlbennettnz** [comment]
v1.6.0

No config except for that one inline rule.

**Comment by ilyavolodin** [comment]
Based on the code of the rule, it looks like this is a bug, and rule should allow it. Also error message is pointing to the wrong location (for keyword instead of the test node).

**Comment by mysticatea** [comment]
I will try to fix this.
